content/blog/2022/09/Vim dap/index.md

```diff
@@ -0,0 +1,213 @@
+Install the vimspector using vim-plug plugin manager.
+- Inside `.vimrc` file:
+- `Plug 'puremourning/vimspector'`
+- `:PlugInstall`
+
+# How to get Camel Debug Adapter server.jar file?
+- I found 2 releases of camel DAP server here - https://github.com/camel-tooling/camel-debug-adapter/tags but it doesn't contain any jar files. To get a jar file either you can compile the source or download from the Maven Central repository. 
+- To download from Maven Central repository
+```sh
+cd /home
+curl -LO https://repo1.maven.org/maven2/com/github/camel-tooling/camel-dap-server/0.4.0/camel-dap-server-0.4.0.jar
+  ```
+# Inside .vimrc file:
+```viml
+  call plug#begin()
+  " The default plugin directory will be as follows:
+  "   - Vim (Linux/macOS): '~/.vim/plugged'
+  "   - Vim (Windows): '~/vimfiles/plugged'
+  "   - Neovim (Linux/macOS/Windows): stdpath('data') . '/plugged'
+  " You can specify a custom plugin directory by passing it as the argument
+  "   - e.g. `call plug#begin('~/.vim/plugged')`
+  "   - Avoid using standard Vim directory names like 'plugin'
+  "
+  "
+  " Use release branch (recommend)
+  Plug 'neoclide/coc.nvim', {'branch': 'release'}
+  Plug 'puremourning/vimspector'
+  call plug#end()
+  
+  " https://github.com/puremourning/vimspector/blob/master/README.md#human-mode
+  let g:vimspector_enable_mappings = 'HUMAN'
+
+  " Vimspector key mappings 
+  nnoremap <Leader>dd :call vimspector#Launch()<CR>
+  nnoremap <Leader>dt :call vimspector#ToggleBreakpoint()<CR>
+  nnoremap <Leader>dc :call vimspector#Continue()<CR>
+  nnoremap <Leader>de :call vimspector#Reset()<CR>
+```
+
+# Configure camel-example Debug Gadget
+Place this content in `cust_camel-debug-adapter.json` in your vimspector directory (path might be different for you)
+```sh
+cd ~/.vim/plugged/vimspector/gadgets/linux/.gadgets.d
+```
+Vimspector adapter configuration:
+```json
+{
+  "adapters": {
+    "cust_camel-debug-adapter": {
+         "command": [
+        "java",
+        "-jar",
+        "/home/camel-dap-server-0.4.0.jar"
+         ]
+    }
+  }
+}
+```
+# Vimspector Config
+ Create a file called `.vimspector.json` in the project root and then place the following content
+
+ Vimspector debugger configuration:
+ ```json
+{
+  "configurations": {
+    "Camel Debug Adapter - Attach": {
+      "adapter": "cust_camel-debug-adapter",
+      "configuration": {
+        "request": "attach",
+        "sourcePaths": [ "${workspaceRoot}/src/main/java" ],
+        "hostName": "localhost",
+        "port": "${JVMDebugPort}",
+        "stepFilters": {
+          "skipClasses": [ "$$JDK" ]
+        }
+      }
+    }
+  }
+}
+```
+# How to attach camel application to vimspector for debugging
+
+* To test the camel application debugging need a working example. Lot of good examples are provided in this repo, [camel-examples](https://github.com/apache/camel-examples).
+
+* To enable Camel Textual Route debugging, you need to launch this examples with the profile `camel.debug`.
```

Comment:
In fact i think that now we can use directly `mvn camel:debug` withotu adding a profile